feat: Implement UI element to annotate YAML files

niallthomson opened the pull request:
#### What this PR does / why we need it:

This PR adds a new markdown UI element that can be used to break down YAML snippets with annotations to call out each important section and what its doing. This is currently being done in several labs but without much consistency and in generally still requiring a lot of effort for the user to pick through the YAML to match the explanations with parts of the file.

This is an example of the markdown used:

```
::yaml{file="manifests/modules/autoscaling/compute/karpenter/scale/deployment.yaml" paths="spec.replicas,spec.template.spec.nodeSelector,spec.template.spec.containers.0.image,spec.template.spec.containers.0.resources"}

1. Initially specifies 0 replicas to run, we'll scale it up later
2. Requires the pods to be scheduled to capacity provisioned by Karpenter by using a node selector that matches our NodePool
3. Uses a simple `pause` container image
4. Requests `1Gi` of memory for each pod
```

Things to note:
1. File is referenced by path
2. The "paths" attribute accepts a comma-separated list of paths to highlight in a similar format to `jq`. You can reference specific fields (`metadata.name`), whole blocks/multiple lines (`spec.template`) and specific items in arrays using numbers (`spec.template.spec.containers.0`) as well as fields of them (`spec.template.spec.containers.0.name`).
3. Optionally you can include a markdown ordered list as the next markdown element and it will correlate these in the order that the "paths" attribute was specified. So item (1) will match the first entry in "paths", (2) will match the second etc.

The above example will render like this:

![Screenshot 2024-06-22 at 7 43 45 PM](https://github.com/aws-samples/eks-workshop-v2/assets/808369/6b6c6e3d-2794-4d9f-8123-470ec62a48ed)

#### Which issue(s) this PR fixes:

N/A

#### Quality checks

- [x] My content adheres to the style guidelines
- [x] I ran `make test module="<module>"` it was successful (see https://github.com/aws-samples/eks-workshop-v2/blob/main/docs/automated_tests.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

svennam92 commented:
@niallthomson I like this particularly being able to use markdown formatting like code blocks in the descriptor text. Can we add your blurb above to the style guide? https://github.com/aws-samples/eks-workshop-v2/blob/main/docs/style_guide.md

niallthomson commented:
@svennam92 based on some feedback I changed the annotations from numbers to letters to better distinguish from the line numbers. Updated the style guide and UI components docs.
